### PR TITLE
Aaron/add clib doc

### PR DESCRIPTION
**Pull request by ambray**
Added document detailing mechanisms for interfacing Python applications with C/C++.

### CODE REVIEW

File: docs/scenarios/clibs.rst [hunk 1]
```diff
@@ -0,0 +1,106 @@
+Interfacing with C/C++ Libraries
+================================
+
+
+ctypes
+------
+
```
Comment (sigmavirus24) [inline docs/scenarios/clibs.rst]: You should mention that ctypes is only available on CPython and only works there. It is not supported by PyPy or any other implementation that I'm aware of.

**Comment by sigmavirus24** [comment]
Any discussion of using C libraries without mentioning CFFI seems incomplete. Would you like to add that discussion @ambray? Most prominent projects that aim to support PyPy, CPython, and other platforms choose CFFI.

**Comment by ambray** [comment]
CFFI section added, and ctypes has been amended to reflect the fact that it is somewhat tethered to CPython.

**Comment by sigmavirus24** [comment]
Thanks @ambray!

**Pull request merged**